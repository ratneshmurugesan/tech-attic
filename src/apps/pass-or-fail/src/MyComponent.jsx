import React from 'react';

export const MyComponent = ({one = null, two = null}) => {
    return (
        <div>
            <h1>React+hooks+testing</h1>
            <p>{one}</p>
            <p>{two}</p>
            <button id="button-one" onClick={() => console.log('clicked')}></button>
        </div>
    )
}
