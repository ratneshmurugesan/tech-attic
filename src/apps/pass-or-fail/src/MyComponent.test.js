import React from 'react';
import { 
    shallow, 
    mount } from 'enzyme';
// import fn from 'jest';

import "./setupTests";

import { MyComponent} from './MyComponent';

const clickFn = jest.fn();
// const mockTryGetValue = jest.fn(() => false);
// const mockTrySetValue = jest.fn(); 
 
// jest.mock('save-to-storage', () => ({   
//   SaveToStorage: jest.fn().mockImplementation(() => ({
//     tryGetValue: mockTryGetValue,
//     trySetValue: mockTrySetValue,
//   })), 
// }));
describe('MyComponent', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<MyComponent debug />);
        expect(component).toMatchSnapshot();
    });
    it('should render correctly', () => {
        const component = shallow(<MyComponent />);
        expect(component).toMatchSnapshot();
    });
    it('should render with child "button" to trigger EVENTS', () => {
        const list = { one: '1', 'two': '2' };
        const component = mount(<MyComponent list={list} />);
        component.find('button#button-one').simulate('keydown', { keyCode: 32 }); // Interacting with child using mount
        expect(component).toMatchSnapshot();
        component.unmount();
    });
    it('should simulate click event using MOCK functions', () => {
        // console.log('clickFn', clickFn);
        const component = shallow(<MyComponent onClick={clickFn} />);
        component.find('button#button-one').simulate('click');

        expect(clickFn).toHaveBeenCalled();
        component.unmount();
    });
    // it('should set storage on save button click', () => {
    //     mockTryGetValue.mockReturnValueOnce(true);
    //     const component = mount(<MyComponent />); 
    //     component.find('button#my-button-three').simulate('click');
    //     expect(mockTryGetValue).toHaveBeenCalled();
    //     console.log(component.debug())
    //     expect(component).toMatchSnapshot();
    //     component.unmount();   
    //   }); 
});
