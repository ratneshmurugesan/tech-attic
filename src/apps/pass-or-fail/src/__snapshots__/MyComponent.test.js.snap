// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly 1`] = `
<div>
  <h1>
    React+hooks+testing
  </h1>
  <p />
  <p />
  <button
    id="button-one"
    onClick={[Function]}
  />
</div>
`;

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div>
  <h1>
    React+hooks+testing
  </h1>
  <p />
  <p />
  <button
    id="button-one"
    onClick={[Function]}
  />
</div>
`;

exports[`MyComponent should render with child "button" to trigger EVENTS 1`] = `
<MyComponent
  list={
    Object {
      "one": "1",
      "two": "2",
    }
  }
>
  <div>
    <h1>
      React+hooks+testing
    </h1>
    <p />
    <p />
    <button
      id="button-one"
      onClick={[Function]}
    />
  </div>
</MyComponent>
`;
